web/src/pages/apd/activities/overview/NameAndFundingSource/NameAndFundingSourceForm.js

```diff
@@ -1,59 +1,105 @@
 import { ChoiceList, TextField } from '@cmsgov/design-system';
 import PropTypes from 'prop-types';
 import React, { Fragment, useCallback } from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import { joiResolver } from '@hookform/resolvers/joi';
 import { connect } from 'react-redux';
 
+import nameFundingSourceSchema from '../../../../../static/schemas/nameAndFundingSource';
 import {
   setActivityName,
   setActivityFundingSource
 } from '../../../../../redux/actions/editActivity';
 
-const NameAndFundingSourceForm = ({
-  index,
-  item: { fundingSource, name },
-  setFundingSource,
-  setName
-}) => {
-  const changeName = useCallback(
-    ({ target: { value } }) => {
-      setName(index, value);
-    },
-    [index, setName]
-  );
+const NameAndFundingSourceForm = ({ 
+    index, 
+    item: { fundingSource, name }, 
+    setFundingSource,
+    setName 
+  }) => {
+    NameAndFundingSourceForm.displayName = 'NameAndFundingSourceForm';
 
-  const changeFundingSource = useCallback(
-    ({ target: { value } }) => {
-      setFundingSource(index, value);
-    },
-    [index, setFundingSource]
-  );
+    const {
+      control,
+      formState: { errors }
+    } = useForm({
+      defaultValues: {
+        name: name,
+        fundingSource: fundingSource
+      },
+      mode: 'onBlur',
+      reValidateMode: 'onBlur',
+      resolver: joiResolver(nameFundingSourceSchema)
+    });
 
-  const choices = ['HIT', 'HIE', 'MMIS'].map(choice => ({
-    checked: fundingSource === choice,
-    label: choice,
-    value: choice
-  }));
+    try {
```

Comment:
What issues were you running into to catch the errors? This might be better as a useEffect if you want it to run when the component first renders (or when state change happens)

Reply:
I'll try this!